lib/plugin.py

```diff
@@ -103,14 +93,22 @@ def do_check_in_credential(self):
     def _make_safeguard_instance(self):
         if not self._safeguard_client_factory:
             self._safeguard_client_factory = SafeguardClientFactory.from_config(self.plugin_configuration)
+
         safeguard = self._safeguard_client_factory.new_instance(
-            access_token=self.access_token,
-            session_access_token=self.token,
-            gateway_username=self.connection.gateway_username,
-            gateway_password=self.connection.gateway_password,
+            access_token=self.access_token, session_access_token=self.token, **self._get_auth_credentials()
         )
         return safeguard
 
+    def _get_auth_credentials(self):
```

Comment:
I am not sure why this is here

Reply:
There's a scenario when [safeguard-password-authentication]/use_credential option is "token". So by checking it before calling self.authorization_username/password we avoid having to deal with the exception thrown by the base class. (In the credential store base class we only provide two options for the use_credential option: gateway and explicit.)

Reply:
That's because the base class only check enum with explicit and gateway keys
Does this mean that the SDK is not so easy to use?
Or this case is an exception and we should handle this in that way